RequestService/RequestService.Handlers/PutUpdateJobDueDateHandler.cs

```diff
@@ -30,24 +31,52 @@ public async Task<PutUpdateJobDueDateResponse> Handle(PutUpdateJobDueDateRequest
                 Outcome = UpdateJobOutcome.Unauthorized
             };
 
+            if (request.DueDate < DateTime.UtcNow)
+            {
+                return response;
+            }
+
             bool hasPermission = await _jobService.HasPermissionToChangeJobAsync(request.JobID.Value, request.AuthorisedByUserID.Value, cancellationToken);
 
             if (hasPermission)
             {
+                var jobDetails = _repository.GetJobDetails(request.JobID.Value);
+
+                if (
+                    (jobDetails == null) || 
+                    (jobDetails.JobSummary.JobStatus == JobStatuses.Done || jobDetails.JobSummary.JobStatus == JobStatuses.Cancelled)
+                    )
+                {
+                    return response;
+                }
+
+                var oldValue = jobDetails.JobSummary.DueDate;
                 var result = await _repository.UpdateJobDueDateAsync(request.JobID.Value, request.AuthorisedByUserID.Value, request.DueDate, cancellationToken);
                 response.Outcome = result;
 
                 if (result == UpdateJobOutcome.Success)
                 {
                     response.Outcome = UpdateJobOutcome.Success;
+
+                    await _repository.UpdateHistory(
+                        requestId: jobDetails.JobSummary.RequestID,
+                        createdByUserId: request.AuthorisedByUserID.Value,
+                        fieldChanged: "Due Date",
+                        oldValue: oldValue.ToString(),
```

Comment:
N.B. These will be UTC times.  Not sure if that will cause any problems downstream...